pydantic_ai_slim/pydantic_ai/agent/__init__.py

```diff
@@ -89,6 +90,8 @@
 S = TypeVar('S')
 NoneType = type(None)
 
+AgentMetadataValue = str | dict[str, str] | Callable[[RunContext[AgentDepsT]], str | dict[str, str]]
```

Comment:
I opted to not introduce a generic `MetadataT` to the Agent, but instead allow strings or dicts, that get serialized into JSON-strings for the OTel attribute.